libusb.go

```diff
@@ -301,13 +301,28 @@ func (libusbImpl) getDeviceDesc(d *libusbDevice) (*DeviceDesc, error) {
 			}
 			descs := make([]InterfaceSetting, 0, len(alts))
 			for _, alt := range alts {
+
+				var extraDescriptors []*RawDescriptor
+				extra := C.GoBytes(unsafe.Pointer(alt.extra), alt.extra_length)
+				for i := 0; i != len(extra); i += int(extra[i]) {
+					block := extra[i : i+int(extra[i])]
+					bType := DescriptorType(block[1])
+
+					extraDescriptors = append(extraDescriptors,
+						&RawDescriptor{
+							Type: bType,
+							Data: block},
+					)
+				}
+
```

Comment:
extract to a helper function, since we're going to use that on interfaces as well as endpoints and configs.